Lib/ufo2ft/filters/dottedCircleFilter.py

```diff
@@ -0,0 +1,255 @@
+            glyphSet = _GlyphSet.from_layer(font)
+
+        self.set_context(font, glyphSet)
+        added_glyph = False
+        dotted_circle = self.check_dotted_circle()
+        if not dotted_circle:
+            dotted_circle = self.draw_dotted_circle(glyphSet)
+            added_glyph = True
+
+        added_anchors = self.check_and_add_anchors(dotted_circle)
+
+        if added_anchors:
+            self.ensure_base(dotted_circle)
+
+        if added_glyph or added_anchors:
+            return [dotted_circle]
+        else:
+            return []
+
+    def check_dotted_circle(self):
+        """Check for the presence of a dotted circle glyph and return its name"""
+        font = self.context.font
+        dotted_circle = next((g.name for g in font if 0x25CC in g.unicodes), None)
+        if dotted_circle:
+            logger.debug("Found dotted circle glyph %s", dotted_circle)
+            return dotted_circle
+
+    def draw_dotted_circle(self, glyphSet):
+        """Add a new dotted circle glyph, drawing its outlines"""
+        font = self.context.font
+        logger.debug("Adding dotted circle glyph")
+        glyph = Glyph(name="uni25CC", unicodes=[0x25CC])
+        pen = glyph.getPen()
+
+        bigradius = (font.info.xHeight - 2 * self.options.margin) / 2
+        littleradius = bigradius / 6
+        left = self.options.sidebearing + littleradius
+        right = self.options.sidebearing + bigradius * 2 - littleradius
+        middleY = font.info.xHeight / 2
+        middleX = (left + right) / 2
+        subangle = 2 * math.pi / self.options.dots
+        for t in range(self.options.dots):
+            angle = t * subangle
+            cx = middleX + bigradius * math.cos(angle)
+            cy = middleY + bigradius * math.sin(angle)
+            circle(pen, (cx, cy), littleradius)
+
+        glyph.setRightMargin(self.options.sidebearing)
+        try:
+            font.addGlyph(glyph)
+        except AttributeError:
+            font.insertGlyph(glyph)
+
+        glyphSet["uni25CC"] = glyph
+        return "uni25CC"
+
+    def check_and_add_anchors(self, dotted_circle):
+        """Check that all mark-attached anchors are present on the dotted
+        circle glyph, synthesizing a position for any missing anchors."""
+        font = self.context.font
+
+        # First we will gather information about all the anchors in the
+        # font at present; for the anchors on marks (starting with "_")
+        # we just want to know their names, so we can match them with
+        # bases later. For the anchors on bases, we also want to store
+        # the position of the anchor so we can average them.
+        all_anchors = {}
+        any_added = False
+        anchorclass = None
+        for glyph in font:
+            width = None
+            try:
+                bounds = glyph.getBounds(font)
+                width = bounds.xMax - bounds.xMin
+            except AttributeError:
+                bounds = glyph.bounds
```

Comment:
`AttributeError` can happen because `bounds` is None, does this not need to be handled by the except arm?